internal/server/v1beta1/search_test.go

```diff
@@ -332,3 +334,235 @@ func TestSuggest(t *testing.T) {
+							Description: "some description",
+							Service:     "test-service",
+							Labels: map[string]string{
+								"entity":    "gotocompany",
+								"landscape": "id",
+							},
+						},
+						},
+					},
+				}
+				as.EXPECT().GroupAssets(ctx, cfg).Return(response, nil)
+			},
+
+			PostCheck: func(resp *compassv1beta1.GroupAssetsResponse) error {
+				expected := &compassv1beta1.GroupAssetsResponse{
+					AssetGroups: []*compassv1beta1.AssetGroup{
+						{
+							GroupFields: []*compassv1beta1.GroupField{
+								{
+									GroupKey:   "resource",
+									GroupValue: "kafka",
+								},
+							},
+							Assets: []*compassv1beta1.Asset{
+								{
+									Id:          "test-resource",
+									Description: "some description",
+									Service:     "test-service",
+									Type:        "test",
+									Labels: map[string]string{
+										"entity":    "gotocompany",
+										"landscape": "id",
+									},
+								},
+							},
+						},
+					},
+				}
+				testutils.AssertEqualProto(t, expected, resp)
+				return nil
+			},
+		},
+		{
+			Description: "should return the requested number of assets",
+			Request: &compassv1beta1.GroupAssetsRequest{
+				Groupby: []string{"resource"},
+				Size:    2,
+			},
+			Setup: func(ctx context.Context, as *mocks.AssetService) {
+				cfg := asset.GroupConfig{
+					GroupBy:        []string{"resource"},
+					Size:           2,
+					Filters:        make(map[string][]string),
+					IncludedFields: []string(nil),
+				}
+
+				results := make([]asset.GroupResult, cfg.Size)
+				asset1 := asset.Asset{
+					Type:    "topic",
+					Service: "kafka",
+					Labels: map[string]string{
+						"landscape": "id",
+						"entity":    "gotocompany",
+					},
+				}
+				kafkaAssets := []asset.Asset{asset1}
+				asset2 := asset.Asset{
+					Type:    "table",
+					Service: "bigquery",
+					Labels: map[string]string{
+						"landscape": "id",
+						"entity":    "gotocompany",
+					},
+				}
+
+				bigqueryAssets := []asset.Asset{asset2}
+
+				results[0] = asset.GroupResult{
+					GroupFields: []asset.GroupField{{
+						GroupKey:   "resource",
+						GroupValue: "kafka",
+					},
+					},
+					Assets: kafkaAssets,
+				}
+
+				results[1] = asset.GroupResult{
+					GroupFields: []asset.GroupField{{
+						GroupKey:   "resource",
+						GroupValue: "bigquery",
+					},
+					},
+					Assets: bigqueryAssets,
+				}
+
+				as.EXPECT().GroupAssets(ctx, cfg).Return(results, nil)
+			},
+			PostCheck: func(resp *compassv1beta1.GroupAssetsResponse) error {
+				expectedSize := 2
+				actualSize := len(resp.AssetGroups)
+				assert.Equal(t, expectedSize, actualSize)
+				return nil
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.Description, func(t *testing.T) {
+			ctx := user.NewContext(context.Background(), user.User{UUID: userUUID})
+
+			logger := log.NewNoop()
+			mockUserSvc := new(mocks.UserService)
+			mockSvc := new(mocks.AssetService)
```

Comment:
Use the `mocks.NewUserSerice` variants. It will set up the expectations assertion in the constructor itself.

Reply:
done